inst/include/common/information.hpp

```diff
@@ -24,16 +25,45 @@ namespace fims {
  */
 template <typename T>
 class Information {
-  typedef fims::FIMSTrait<T>::variable_t variable_t;
-  typedef fims::FIMSTrait<T>::real_t real_t;
-  std::vector<variable_t*> parameters;  // list of all estimated parameters
-  std::vector<variable_t*>
+  public:
+  static std::shared_ptr<Information<T> > fims_information;
+  // typedef fims::FIMSTraits<T>::variable_t variable_t;
+  // typedef fims::FIMSTraits<T>::real_t real_t;
+  std::vector<T*> parameters;  // list of all estimated parameters
+  std::vector<T*>
       random_effects_parameters;  // list of all random effects parameters
-  std::vector<variable_t*>
+  std::vector<T*>
       fixed_effects_parameters;  // list of all fixed effects parameters
+  
+  std::map<uint32_t, std::shared_ptr<fims::RecruitmentBase<T> > > recruitment_models;
+  std::map<uint32_t, std::shared_ptr<fims::SelectivityBase<T> > > selectivity_models;
+  
+static std::shared_ptr<Information<T> > get_instance(){
```

Comment:
per Google style guides, function names should be CamelCase e.g. GetInstance()

Reply:
Class method and function names throughout the FIMS codebase use lowercase/underscore conventions. Class names are using CamelCase, should we refactor the whole codebase?